src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy

```diff
@@ -74,16 +70,17 @@ class IqPolicyEvaluatorUtil
 
       return evaluationResult
     }
-    catch (IqNetworkException e) {
+    catch (IqClientException e) {
       return handleNetworkException(iqPolicyEvaluator.failBuildOnNetworkError, e, listener, run)
     }
   }
 
-  private static handleNetworkException(final boolean failBuildOnNetworkError, final IqNetworkException e,
-                                    final TaskListener listener, final Run run)
+  private static handleNetworkException(final Boolean failBuildOnNetworkError, final IqClientException e,
+                                        final TaskListener listener, final Run run)
   {
-    if (failBuildOnNetworkError) {
-      throw e.cause
+    def isNetworkError = isNetworkError(e)
+    if (!isNetworkError || isNetworkError && failBuildOnNetworkError) {
```

Comment:
seems like we need some braces here on the second part of that conditional:
( !isNetworkError || ( isNetworkError && failBuildOnNetworkError) )

Reply:
For readability?

Reply:
Operator precedence has && evaluated before ||. This got me for a few minutes too.

Reply:
!isNetworkError is a gate to the second part.